### PR TITLE
refactor: use @puppeteer/browsers for install

### PR DESCRIPTION
**Pull request by OrKoN**
The CL rewrite install.ts to use the @puppeteer/browsers package. It should keep all existing functionality and have no observable changes.

### CODE REVIEW

File: packages/puppeteer/src/node/install.ts [hunk 1]
```diff
@@ -37,167 +37,56 @@ const supportedProducts = {
...
-    }
-
-    return browserFetcher
-      .download(revisionInfo.revision, onProgress)
-      .then(() => {
-        return browserFetcher.localRevisions();
-      })
-      .then(onSuccess)
-      .catch(onError);
+  const platform = detectBrowserPlatform();
```
Comment (OrKoN) [inline packages/puppeteer/src/node/install.ts]: TODO: special mode for mac ARM

File: packages/puppeteer/src/node/install.ts [hunk 2]
```diff
@@ -37,168 +39,67 @@ const supportedProducts = {
...
-    }
-
-    return browserFetcher
-      .download(revisionInfo.revision, onProgress)
-      .then(() => {
-        return browserFetcher.localRevisions();
-      })
-      .then(onSuccess)
-      .catch(onError);
+  // TODO: remove once Mac ARM is enabled by default for Puppeteer.
```
Comment (Lightning00Blade) [inline packages/puppeteer/src/node/install.ts]:
```suggestion
  // TODO: remove once Mac ARM is enabled by default for Puppeteer https://github.com/puppeteer/puppeteer/issues/9630.
```

File: test/installation/.mocharc.cjs [hunk 3]
```diff
@@ -19,5 +19,5 @@ module.exports = {
   spec: ['build/**/*.spec.js'],
   timeout: '240000ms',
   // Parallel processing fails on other package managers due to caching.
-  parallel: (process.env['PKG_MANAGER'] ?? 'npm') === 'npm',
+  // parallel: (process.env['PKG_MANAGER'] ?? 'npm') === 'npm',
```
Comment (Lightning00Blade) [inline test/installation/.mocharc.cjs]: Is this still needed? Either uncomment or remove.

**Review by Lightning00Blade — APPROVED** [review]

**Comment by OrKoN** [comment]
I will work on this more: currently not ready to be landed.

**Review by Lightning00Blade — APPROVED** [review]

**Comment by sabberworm** [comment]
AFAICT, this change causes `PUPPETEER_DOWNLOAD_HOST`/`configuration.downloadHost` to no longer be respected.
Flag is still documented, however…
It looks like this can be fixed by passing it as a third argument to `downloadUrls[browser]` in https://github.com/puppeteer/puppeteer/blob/327febe2c9c44bdd48a3e674f2a3894c40ca131e/packages/browsers/src/fetch.ts#L143
However, this would still be a breaking change in case of chromium which previously appended the `/chromium-browser-snapshots` prefix to the configured download URL and now expects it to contain it. However, I welcome this change for clarity, consistency and versatility.

**Comment by sabberworm** [comment]
Yes, I use a private mirror.

**Pull request merged**